### PR TITLE
[ARM] `az bicep publish`: Add parameter `--force` to allow overwriting an existing Bicep module version

### PR DESCRIPTION
**Pull request by SimonWahlin**
**Related command**
`az bicep`

**Description**
`bicep publish` has a new parameter `--force` in version v0.17.1
This PR addd support for that parameter.

For details, please see https://github.com/Azure/bicep/issues/10621

**Testing Guide**
```
az bicep publish --file module.bicep --target "br:<registry>/module:1"
# will publish module to <registry>

az bicep publish --file module.bicep --target "br:<registry>/module:1"
# will show error:
# The module "br:<registry>.azurecr.io/module:1" already exists in registry. Use --force to overwrite the existing module.

az bicep publish --file module.bicep --target "br:<registry>/module:1" --force
# will overwrite existing module
```
**History Notes**
[ARM] `az bicep publish`: Add new parameter `--force` to allow overwriting existing module

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
ARM

**Comment by alex-frankel** [comment]
@shenglol / @majastrz  - can you also take a look at this one?

**Review by shenglol — APPROVED** [review]
:shipit:

File: src/azure-cli/azure/cli/command_modules/resource/_params.py [hunk 1]
```diff
@@ -695,6 +695,7 @@ def load_arguments(self, _):
                                                       help="The target location where the Bicep module will be published."))
         c.argument('documentationUri', arg_type=CLIArgumentType(options_list=['--documentationUri', '-d'],
                                                                 help="The documentation uri of the Bicep module."))
+        c.argument('force', arg_type=bicep_force_type, help="Allows overwriting an existing Bicep module version.")
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/_params.py:698]: Please use the first-person voice in the help message
```suggestion
        c.argument('force', arg_type=bicep_force_type, help="Allow overwriting an existing Bicep module version.")
```

Comment (SimonWahlin) [reply src/azure-cli/azure/cli/command_modules/resource/_params.py]: Done.
Should I change the wording for  --force on decompile and restore as well?

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/_params.py]: Could you give a specific example of what you want to change?

Comment (SimonWahlin) [reply src/azure-cli/azure/cli/command_modules/resource/_params.py]: There are three commands that has a --force parameter (`az bicep publish`, `az bicep decompile` and `az bicep restore`). The help message for the force parameters on decompile and restore uses the word 'Allows'

**Review by zhoxing-ms — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**